dnf-json

```diff
@@ -305,10 +312,15 @@ class DnfJsonRequestHandler(BaseHTTPRequestHandler):
                 arguments = call["arguments"]
                 repos = arguments.get("repos", {})
                 arch = arguments["arch"]
-                self.cache_dir = arguments["cachedir"]
-                cache_state = CacheState.load_cache_state_from_disk(self.cache_dir)
                 module_platform_id = arguments["module_platform_id"]
 
+                # If dnf-json is run as a service, we don't want users to be able to set the
+                # cache
+                self.cache_dir = os.environ.get("OVERWRITE_CACHE_DIR", arguments.get("cachedir", ""))
```

Comment:
BTW, maybe you know why, but I'm wondering, why do we want this to be a parameter in the first place ? What is the upside of having the cache being different for different requests ?

Reply:
I think I do remember having a conversation at some point with Ondrej about the cache folders. Something about the fact that requests sharing the cache folder but having different target architecture are making dnf download everything twice, which slows the process. But I'm not sure if I remember correctly.

Reply:
Ah then we probably want a cache dir per architecture. I do tihnk it's better if dnf-json sets it's own cache rather than users specifying it too I agree. Not sure if there's an argument against it.

Reply:
Ah i think the idea is to be able to set the cache dir in 1 place (composer's CACHE_DIRECTORY), and have that be valid everywhere. CACHE_DIRECTORY is also set in dnf-json's systemd unit file but then not used anywhere. It's all a little confusing tbh, but I think we should keep the ability to set it  in the request.